shared/helpers/assertRevert.js

```diff
@@ -27,8 +27,8 @@ module.exports = web3 => {
     }
   }
 
-  async function assertRevertGeth(request, reason) {
-    const tx = request.params[0]
+  async function assertRevertGeth(truffleInstance, requestOrTx, reason) {
```

Comment:
It seems quite odd to require the truffle-contract instance here (and it deviates from a lot of other `assertRevert` code we have across the org), why was this needed?

Reply:
For some odd reason, extracting the tx and sending it via web3.eth.sendTransaction does not produce a revert error with the correct reason. Sending it via the truffle object does not have that problem.

Reply:
Hmm, this is awkward. It reverts with a different reason when sending through with web3?

Reply:
It reverts with no reason, just a plain "reverted" error.